app/controllers/sign_up/completions_controller.rb

```diff
@@ -84,7 +84,6 @@ def analytics_attributes(page_occurence)
         ialmax: resolved_authn_context_result.ialmax?,
         service_provider_name: decorated_sp_session.sp_name,
         sp_session_requested_attributes: sp_session[:requested_attributes],
-        sp_request_requested_attributes: service_provider_request.requested_attributes,
```

Comment:
tracing this back looks like it should come from

https://github.com/18F/identity-idp/blob/449ce683b1921708f670b6d5be281c96ab7c90e9/app/services/service_provider_request_handler.rb#L68
which comes from either

https://github.com/18F/identity-idp/blob/9aadb9818fe2b5a63a81253db5a8d6a361d7b4a3/app/models/federated_protocols/oidc.rb#L29-L31

or

https://github.com/18F/identity-idp/blob/9aadb9818fe2b5a63a81253db5a8d6a361d7b4a3/app/models/federated_protocols/saml.rb#L33-L40

which don't *seem* like they should be as empty as they are?

Reply:
But the root of the issue is that at this point, [`ApplicationController#service_provider_request`](https://github.com/18F/identity-idp/blob/449ce683b1921708f670b6d5be281c96ab7c90e9/app/controllers/application_controller.rb#L204-L206) will always be `NullServiceProviderRequest`, because [`uuid` is blank](https://github.com/18F/identity-idp/blob/449ce683b1921708f670b6d5be281c96ab7c90e9/app/services/service_provider_request_proxy.rb#L15) since there is no `request_id` query parameter at this controller.

Reply:
oh that explains a lot

Reply:
maybe a next PR could clean up that helper? I know in at least one place use do pass `request_id` as a URL param for emails, but maybe it shouldn't be in a shared controller like it is now

Reply:
Yeah, that's part of the trickiness, since there are a few places we do expect `request_id` query parameter to be supported (the entrypoints like password reset, sign-in), so we might rely on it there. But I'd think the way it _should_ work is that we immediately store it all in the session and then refer to `sp_session` from that point forward.